quantization/src/encoded_vectors_binary.rs

```diff
@@ -120,30 +121,38 @@ impl<TStorage: EncodedStorage> EncodedVectorsBin<TStorage> {
     }
 
     fn calculate_metric(&self, v1: &[BitsStoreType], v2: &[BitsStoreType]) -> f32 {
-        let xor_product = Self::xor_product(v1, v2);
-
         // Dot product in a range [-1; 1] is approximated by NXOR in a range [0; 1]
+        // L1 distance in range [-1; 1] (alpha=2) is approximated by alpha*XOR in a range [0; 1]
+        // L2 distance in range [-1; 1] (alpha=2) is approximated by alpha*sqrt(XOR) in a range [0; 1]
```

Comment:
Why `sqrt(XOR)` here?

Reply:
well sqrt is there in the L2 definition, if you are not taking the sqrt at a later step you are actually calculating L2^2.

I had actually confirmed these approximations through tests, but decided to forgoe te actual value approximations for calculation simplicity. alpha or sqrt does not change the order.